### DISCUSSION TITLE
i-need-your-help-with-intel-quick-sync-benchmarking/

### DISCUSSION THREAD

**Discussion by giscus** [ironicbadger/blog.ktz.me#223] [Announcements]
# i-need-your-help-with-intel-quick-sync-benchmarking/

Please help me benchmark as many Intel CPUs Quick Sync transcoding performance as possible!

https://blog.ktz.me/i-need-your-help-with-intel-quick-sync-benchmarking/

**Comment by BeardedTek** [comment]
I wonder if this will work on an Ark A770?  I just picked one up and will get it installed tomorrow.

**Comment by dukotron** [comment]
I ran this test on several different configurations, all of which were given 30.25GiB of RAM and 16 vCPUs (aka max available to match up with host):

Proxmox Host (All VMS/LXCs turned off)
CPU           TEST                      FILE                                         BITRATE      TIME        AVG_FPS  AVG_SPEED  AVG_WATTS
i7-10700K  h264_1080p_cpu  ribblehead_1080p_h264         18952 kb/s  31.706s   114.55      3.91x            N/A
i7-10700K  h264_1080p         ribblehead_1080p_h264         18952 kb/s  13.131s    262.87     9.06x            7.73
i7-10700K  h264_4k               ribblehead_4k_h264                46881 kb/s  49.527s   68.55       2.42x             7.54
i7-10700K  hevc_8bit             ribblehead_1080p_hevc_8bit  14947 kb/s   41.261s   83.46       2.91x            8.91
i7-10700K  hevc_4k_10bit     ribblehead_4k_hevc_10bit       44617 kb/s  151.554s  23.04       .80x             10.20

Ubuntu LXC with Passthrough
CPU           TEST                      FILE                                         BITRATE      TIME        AVG_FPS  AVG_SPEED  AVG_WATTS
i7-10700K  h264_1080p_cpu  ribblehead_1080p_h264         18952 kb/s  31.654s   116.00      3.84x           N/A
i7-10700K  h264_1080p         ribblehead_1080p_h264         18952 kb/s  13.427s    256.11     8.90x           7.61
i7-10700K  h264_4k               ribblehead_4k_h264                46881 kb/s  49.234s    69.24      2.48x           7.56
i7-10700K  hevc_8bit             ribblehead_1080p_hevc_8bit  14947 kb/s   41.552s    83.32     2.96x            8.89
i7-10700K  hevc_4k_10bit     ribblehead_4k_hevc_10bit       44617 kb/s   152.183s  23.04     .80x             10.19

Ubuntu VM with Intel GVT-g V5_4
CPU           TEST                      FILE                                         BITRATE      TIME        AVG_FPS  AVG_SPEED  AVG_WATTS
i7-10700K  h264_1080p_cpu  ribblehead_1080p_h264         18952 kb/s  33.046s   113.75      3.77x           N/A
i7-10700K  h264_1080p         ribblehead_1080p_h264         18952 kb/s  14.869s    228.76     7.91x
i7-10700K  h264_4k               ribblehead_4k_h264                46881 kb/s  53.412s    63.96      2.27x
i7-10700K  hevc_8bit             ribblehead_1080p_hevc_8bit  14947 kb/s   43.473s    78.50     2.74x
i7-10700K  hevc_4k_10bit     ribblehead_4k_hevc_10bit       44617 kb/s   156.844s  21.94     .78x

Ubutnu VM with Intel GVT-g V5_8
CPU           TEST                      FILE                                         BITRATE      TIME        AVG_FPS  AVG_SPEED  AVG_WATTS
i7-10700K  h264_1080p_cpu  ribblehead_1080p_h264         18952 kb/s  33.172s   114.57      3.75x           N/A
i7-10700K  h264_1080p         ribblehead_1080p_h264         18952 kb/s  14.926s    228.30     7.92x
i7-10700K  h264_4k               ribblehead_4k_h264                46881 kb/s  53.514s    63.84      2.26x
i7-10700K  hevc_8bit             ribblehead_1080p_hevc_8bit  14947 kb/s   43.464s    79.08     2.77x
i7-10700K  hevc_4k_10bit     ribblehead_4k_hevc_10bit       44617 kb/s   157.107s  21.94     .78x

To my surprise, all of the results were very close to each other. Almost no performance penalty for passthrough or GVT-g. I am particularly surprised that GVT-g V5_4 has no noticable difference to V5_8, given that they are by definition giving different amounts of juice to the VM.

It makes me wonder how you came to this conclusion: https://blog.ktz.me/why-i-stopped-using-intel-gvt-g-on-proxmox/

Or is this benchmark not a good tool to benchmark these differences, but only between CPUs? I can't really explain these results, maybe you can make sense of them.

**Comment by alcatraz-zz** [comment]
Will this also benchmark quicksync encoding performance? It seems it's only doing decode. Hmm...

**Reply by ironicbadger** [reply]
PRs welcome 👍

**Comment by gh975223-create** [comment]
does not work for me:

Checking for test video files...
All test files present.
Pulling Jellyfin container...
permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.50/images/create?fromImage=docker.io%2Fjellyfin%2Fjellyfin&tag=latest": dial unix /var/run/docker.sock: connect: permission denied
Starting Jellyfin QSV test container (attempt 1/3)...
  Failed to start container
Starting Jellyfin QSV test container (attempt 2/3)...
  Failed to start container
Starting Jellyfin QSV test container (attempt 3/3)...
  Failed to start container
ERROR: Failed to start Jellyfin container after 3 attempts
Try manually: docker rm -f jellyfin-qsvtest

**Reply by ironicbadger** [reply]
Looks like some kind of environment issue. Are you docker? Podman? Does the user you tried to run the script as have docker group permissions?
